internal/corazawaf/logger.go

```diff
@@ -1,70 +1,49 @@
 // Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
 // SPDX-License-Identifier: Apache-2.0
 
+//go:generate go run ./logger/generator/main.go
+
 package corazawaf
 
 import (
-	"fmt"
 	"io"
 	"log"
 
+	"github.com/corazawaf/coraza/v3/internal/corazawaf/internal/logger"
 	"github.com/corazawaf/coraza/v3/loggers"
 )
 
-// DebugLogger is a logger that logs to the standard logger
 type stdDebugLogger struct {
 	io.Closer
-	logger *log.Logger
-	Level  loggers.LogLevel
+	logger.DelegateLogger
+	l *log.Logger
 }
 
 var _ loggers.DebugLogger = (*stdDebugLogger)(nil)
 
-func (l *stdDebugLogger) formatLog(level loggers.LogLevel, message string, args ...interface{}) {
-	if l.Level >= level {
-		l.logger.Printf("[%s] %s", level.String(), fmt.Sprintf(message, args...))
```

Comment:
IIUC the only difference among loggers in the generated code corresponds to this `[%s] ` for the level string. It seems it can be more simply

```
func logLevelPrefix(l loggers.LogLevel) string {
switch (l) {
  case loggers.LogLevelInfo: return "[INFO] "
  case loggers.LogLevelWarn: return "[WARN] "
// etc
}
```

```
if l.Level >= level {
  l.logger.Print(logLevelPrefix(level))
  l.logger.Printf(message, args...)
}
```

Reply:
BTW interesting comment on print vs printf

https://go-review.googlesource.com/c/go/+/457435/comments/b3f2a0c1_2aa2b067